### PR TITLE
Fix GDScript analyzer error when instantiating EditorPlugins.

### LINKED ISSUE #73525: EditorExportPlugin raises error everytime you run the game with F5 because of "abstract native class"
### Godot version

4.0 rc2

### System information

Manjaro Linux using i3, Foward+

### Issue description

I've added an ExportPlugin to my addon and everytime I run the game in the editor I know get this error:
```
SCRIPT ERROR: Parse Error: Class "eh_UtilitiesExportPlugin" cannot be constructed as it is based on abstract native class "EditorExportPlugin".
```

I even tried to guard it with an `Engine.is_editor_hint()` call but it doesn't work, I still get the same error even though line 54 shouldn't be executing:
image

Weirdly enough, this guard works fine in other parts of the project, but fails here.

### Steps to reproduce

Just create a script that extends `EditorExportPlugin`, and try to create an instance of it in an `EditorPlugin` script

### Minimal reproduction project

See "Steps to Reproduce"

### PR DESCRIPTION
**Pull request by MikeSchulze**
Continue from originally https://github.com/godotengine/godot/pull/74025

* _Bugsquad edit:_ Fixes #73525.

### CODE REVIEW

**Comment by AThousandShips** [comment]
Waiting to confirm that the original author has indeed abandoned their PR

**Comment by dalexeev** [comment]
@MikeSchulze Since the original author doesn't respond for several days, feel free to continue their work.

**Comment by MikeSchulze** [comment]
@dalexeev done

**Comment by MikeSchulze** [comment]
Ready for review

File: core/object/class_db.cpp [hunk 1]
```diff
@@ -671,6 +671,19 @@ bool ClassDB::can_instantiate(const StringName &p_class) {
...
 
+bool ClassDB::is_abstract(const StringName &p_class) {
+	OBJTYPE_RLOCK;
+
+	ClassInfo *ti = classes.getptr(p_class);
+	if (!ti) {
+		if (!ScriptServer::is_global_class(p_class)) {
+			ERR_FAIL_V_MSG(false, "Cannot get class '" + String(p_class) + "'.");
+		}
+		return false;
```
Comment (dalexeev) [inline core/object/class_db.cpp:682]:
```suggestion
		String path = ScriptServer::get_global_class_path(p_class);
		Ref<Script> scr = ResourceLoader::load(path);
		return scr.is_valid() && scr->is_valid() && scr->is_abstract();
```

I'm not sure why `ClassDB` checks `ScriptServer`, but this is also present in `is_virtual()` and `can_instantiate()` (with the last condition inverted).

**Review by dalexeev — COMMENTED** [review]
Looks good to me and it fixes the bug.

There are some inconsistencies between `is_abstract()` and `is_virtual()`. I think in the future we should remove `#ifdef TOOLS_ENABLED` code and the `!ti->disabled` check from `is_virtual()`, but for now let's leave it as is.

@MikeSchulze Please squash commits into one according to our [PR workflow](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase). Make sure you are the commit author and @baptr is a co-author (or vice versa).

**Comment by MikeSchulze** [comment]
@dalexeev done

**Review by dalexeev — APPROVED** [review]
Note that you are still not the commit author:



Also please do not use GitHub links in commit messages in the future, see [CONTRIBUTING.md](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#format-your-commit-messages-with-readability-in-mind):

> If your commit fixes a reported issue, please include it in the _description_ of the PR (not in the title, or the commit message) using one of the [GitHub closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) such as "Fixes #1234". This will cause the issue to be closed automatically if the PR is merged. Adding it to the commit message is easier, but adds a lot of unnecessary updates in the issue distracting from the thread.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**